### PR TITLE
spec: refactor spec to new API

### PR DESCRIPTION
**Pull request by marbar3778**
## Description 

Refactor spec to adhere to new API. 

This can be considered a long-lived branch, and we amend it as we through the refactor with our findings

### CODE REVIEW

File: module/x/peggy/spec/04_messages.md [hunk 1]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
 - Failure to build a batch of transactions.
 - If the orchestrator address is not present in the validator set
 
-### MsgConfirmBatch
-
-When a `MsgRequestBatch` is observed, validators need to sign batch request to signify this is not a maliciously created batch and to avoid getting slashed. 
+### MsgSubmitConfirm
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:68]: This should be `MsgConfirmEthereumEvent`

File: module/x/peggy/spec/04_messages.md [hunk 2]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
-- If a none validator address or delegated address 
-- If the counter chain address is empty or incorrect.
-- If counter chain address fails signature validation
-- If the signature was already presented in a previous message
+<!-- +++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L79-84 -->
 
-### MsgConfirmLogicCall
+```proto
+  message MsgSubmitClaim {
+    google.protobuf.Any confirm = 1 [
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:76]: wondering if this should be an array instead

File: module/x/peggy/spec/04_messages.md [hunk 3]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
+        (cosmos_proto.accepts_interface) = "EthereumClaim"];
+    string signer = 2;
+  }
+```
 
-When a logic call request has been made, it needs to be confirmed by the bridge validators. Each validator has to submit a confirmation of the logic call being executed.
+There are three types of confirmations a validator can submit.
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:82]:
```suggestion
There are three types of event confirmations a validator can submit.
```

File: module/x/peggy/spec/04_messages.md [hunk 4]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
 
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L155-161
+- `CONFIRM_TYPE_VALSET`
+  - When the peggy daemon witnesses a complete validator set within the peggy module, the validator submits a signature of a message containing the entire validator set.
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:85]: > peggy daemon witnesses a complete validator set within the peggy module

what does "witnessing a complete validator set" mean here? can you add more details?

File: module/x/peggy/spec/04_messages.md [hunk 5]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
+```
 
-When a logic call request has been made, it needs to be confirmed by the bridge validators. Each validator has to submit a confirmation of the logic call being executed.
+There are three types of confirmations a validator can submit.
 
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L155-161
+- `CONFIRM_TYPE_VALSET`
+  - When the peggy daemon witnesses a complete validator set within the peggy module, the validator submits a signature of a message containing the entire validator set. 
+- `CONFIRM_TYPE_LOGIC`
+  - When a logic call request has been made, it needs to be confirmed by the bridge validators. Each validator has to submit a confirmation of the logic call being executed.
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:87]: > logic call request has been made

By whom, where to?

> it needs to be confirmed by the bridge validators

complete: `... by (action performed).`

File: module/x/peggy/spec/04_messages.md [hunk 6]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
+- `CONFIRM_TYPE_BATCH`
+  - When a `MsgRequestBatch` is observed, validators need to sign batch request to signify this is not a maliciously created batch and to avoid getting slashed.
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:89]:
```suggestion
  - When a `MsgRequestBatch` is observed on Cosmos, validators need to sign a batch request to signal this is not a maliciously created batch and thus avoid getting slashed.
```

File: module/x/peggy/spec/04_messages.md [hunk 7]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
-- If the creation of attestation fails
+There are four types of claims:
 
-### MsgWithdrawClaim
-
-When a user requests a withdrawal from the peggy contract a event will omitted by the counter party chain. This event will be observed by a bridge validator and submitted to the gravity module.
-
-
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L187-193
+- `CLAIM_TYPE_DEPOSIT`
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:110]: explain

File: module/x/peggy/spec/04_messages.md [hunk 8]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
+- `CLAIM_TYPE_WITHDRAW`
+  - When a user requests a withdrawal from the peggy contract a event will omitted by the counter party chain. This event will be observed by a bridge validator and submitted to the gravity module.
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:112]: > When a user requests a withdrawal from the peggy contract

withdrawal of funds? what's the purpose of this withdrawal? just to stop participating in the bridge relaying process?

File: module/x/peggy/spec/04_messages.md [hunk 9]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
+  
+- `CLAIM_TYPE_ERC20_DEPLOYED`
+  - This message allows the cosmos chain to learn information about the denom from the counter party chain.
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:115]: > This message allows the cosmos chain to learn information

what type of information? just the contract address and the ERC20 details (symbol, name, decimals, etc)?

File: module/x/peggy/spec/04_messages.md [hunk 10]
```diff
@@ -82,83 +65,59 @@ This message will fail if:
...
+- `CLAIM_TYPE_LOGIC_CALL_EXECUTED`
+  - This informs the chain that a logic call has been executed. This message is submitted by bridge validators when they observe a event containing details around the logic call.
```
Comment (fedekunze) [inline module/x/peggy/spec/04_messages.md:117]: briefly explain what a logic call means

> observe an event containing details around the logic call

which types of details?

**Review by fedekunze — COMMENTED** [review]
Proposed changes LGTM, thanks @marbar3778! A few additional details for further clarification would be great.

I also noticed that we use both peggy and gravity to refer to the module. Ideally, we should stick to the latter.

File: module/x/peggy/spec/04_messages.md (hunk 2 above)
Comment (marbar3778) [reply module/x/peggy/spec/04_messages.md:76]: I can see this as being beneficial and potentially cost saving.

Comment (jkilpatr) [reply module/x/peggy/spec/04_messages.md:76]: +1 this should be an array. Orchestrators catching up after a period of downtime may need to submit hundreds of claims at a time.

**Review by jkilpatr — APPROVED** [review]

**Comment by fedekunze** [comment]
@marbar3778 would love to have this merged as part of the documentation for the refactor. Is there anything I can do to help you unblock this PR?

File: module/x/peggy/spec/04_messages.md (hunk 1 above)
Comment (marbar3778) [reply module/x/peggy/spec/04_messages.md]: was this changed in your work?

**Pull request closed without merging**